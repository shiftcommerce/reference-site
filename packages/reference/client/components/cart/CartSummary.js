// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Libs
import { calculateCartSummary } from '../../lib/calculateCartSummary'
import { fixedPrice } from '../../lib/fixedPrice'

// Objects
import Button from '../../objects/Button'

class CartSummary extends Component {
  shippingText () {
    if (typeof window === 'undefined' || window.location.pathname !== '/checkout') {
      // TODO: make this dynamic in future
      return <>
        <a>Estimated shipping cost: &pound;3.45</a>
      </>
    } else {
      return <>
         <a>Shipping cost:</a>
       </>
    }
  }

  render () {
    const totals = calculateCartSummary(this.props.cart)

    return (
      <>
      <section className='c-cart-summary'>
        <dl>
          <dt><a>Total Price:</a></dt>
          <dd><a>&pound;{fixedPrice(totals.subTotal)}</a></dd>
        </dl>
        <dl>
          <dt>{ this.shippingText() }</dt>
        </dl>
        <dl>
          <dt><a className='u-bold'>TOTAL:</a></dt>
          <dd><a className='u-bold'>&pound;{fixedPrice(totals.total)}</a></dd>
        </dl>
        <dl>
          <a>* Including VAT</a>
        </dl>
      </section>
      <section className='c-cart-summary__buttons'>
        <Button
          aria-label='Continue Shopping'
          label='Continue shopping'
          size='lrg'
          status='grey'
          className='c-cart-summary__buttons--continue'
          type='button'
        />
        <Link href='/checkout' as=''>
          <Button
            aria-label='Place Order'
            label='Continue to payment'
            size='lrg'
            className='c-cart-summary__buttons--proceed'
            status='positive'
            type='button'
          />
        </Link>
      </section>
      </>
    )
  }
}

export default CartSummary
