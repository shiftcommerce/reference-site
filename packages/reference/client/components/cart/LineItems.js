// Libraries
import { Component } from 'react'
import Link from 'next/link'
import { fixedPrice } from '../../lib/fixedPrice'

// Objects
import Image from '../../objects/Image'

class LineItems extends Component {
  // TODO: extract this out into it's own service class
  renderOptions (lineItem) {
    let availableQuantity = Array.from(Array(Math.min(lineItem.stockAvailableLevel, 11)).keys())
    availableQuantity.shift()

    return <select value={lineItem.quantity} onChange={this.props.updateQuantity} data-variant={lineItem.sku}>
      {
        availableQuantity.map(index => {
          return <option key={index} value={index} aria-setsize={index} aria-posinset={index + 1}>
            {index}
          </option>
        })
      }
    </select>
  }

  renderButtonsAndTotal (lineItem) {
    return <>
      <div className='o-grid-item5__lineitems'>
        <div className='c-line-items__buttons'>
          <div className='c-line-items__anchor'>
            <a className='c-line-items__delete' data-variant={lineItem.sku} onClick={this.props.deleteItem} >
              Delete
            </a>
          </div>
        </div>
        <div className='c-line-items__total-wrapper'>
          <a className='c-line-items__total'>&pound;{fixedPrice(lineItem.price * lineItem.quantity)}</a>
        </div>
      </div>
    </>
  }

  renderTitle (lineItem) {
    return <>
      <div className='o-grid-item2__lineitems'>
        <div className='c-line-items__details'>
          <h4 className='c-line-items__details-title  u-bold'>
            {lineItem.title}
          </h4>
          <div className='c-line-items__details-sku'>
            <span>
              {lineItem.sku}
            </span>
          </div>
        </div>
      </div>
    </>
  }

  renderParams (lineItem) {
    return <>
      <div className='o-grid-item3__lineitems'>
        <div className='c-line-items__colour  u-bold'>
          <span>Colour</span>
        </div>
        <div className='c-line-items__size  u-bold'>
          <span>Option</span>
        </div>
        <div className='c-line-items__quantity u-bold'>
          <span>Quantity</span>
        </div>
      </div>
      <div className='o-grid-item4__lineitems'>
        <span className='c-line-items__space'>Grey</span>
        <span className='c-line-items__space'>{lineItem.variant}</span>
        <span className='c-line-items__space'>{ this.renderOptions(lineItem) }</span>
      </div>
    </>
  }

  renderLineItems () {
    const cartData = this.props.cart.lineItems.map((lineItem, index) =>
      <div className='o-grid-container__lineitems' key={lineItem.slug}>
        <div className='o-grid-item1__lineitems'>
          <Link href={`/slug?slug=${lineItem.slug}`} as={`${lineItem.canonical_path}`} key={lineItem.slug}>
            <Image className='c-line-items__image' src={lineItem.imageUrl} alt={lineItem.title} key={lineItem.slug} aria-label={lineItem.title} />
          </Link>
        </div>
        { this.renderTitle(lineItem) }
        { this.renderParams(lineItem) }
        { this.renderButtonsAndTotal(lineItem) }
      </div>
    )
    return cartData
  }

  render () {
    const cart = this.props.cart

    if (cart.totalQuantity === 0) {
      return null
    } else {
      return <div className='c-line-items'>
        { this.renderLineItems() }
      </div>
    }
  }
}

export default LineItems
