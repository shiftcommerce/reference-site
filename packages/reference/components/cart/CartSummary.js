// Libraries
import { Component } from 'react'
import { calculateCartSummary } from '../../lib/calculateCartSummary'

class CartSummary extends Component {
  render () {
    const { subTotal, total, tax, discount } = calculateCartSummary(this.props.cart)

    return (
      <section className='c-cart-summary'>
        <dl>
          <dt> Sub total: </dt>
          <dd> &pound;{ subTotal } </dd>
        </dl>
        <dl>
          <dt> Tax applied: </dt>
          <dd className='u-text-color--red'> +&pound;{ tax } </dd>
        </dl>
        <dl>
          <dt> Amount you saved on this purchase: </dt>
          <dd className='u-text-color--green'> -&pound;{ discount } </dd>
        </dl>
        <dl>
          <dt> Total amount to be paid: </dt>
          <dd> <b>&pound;{ total }</b> </dd>
        </dl>
      </section>
    )
  }
}

export default CartSummary
