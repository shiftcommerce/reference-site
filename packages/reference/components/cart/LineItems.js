import { Component } from 'react'
import Link from 'next/link'

// Objects
import Image from '../../objects/Image'

export default class LineItems extends Component {
  renderOptions (lineItem) {
    return <select value={lineItem.quantity} onChange={this.props.updateQuantity} data-variant={lineItem.sku}>
      {
        [ ...Array(Math.max(lineItem.quantity, 11)) ].map((key, index) =>
          <option value={index} key={index} aria-setsize={index} aria-posinset={index + 1}>
            { index }
          </option>
        )
      }
    </select>
  }

  renderLineItems () {
    const cartData = this.props.cart.lineItems.map((lineItem, index) =>
      <tr key={index} className='c-line-items__column'>
        <td aria-label={lineItem.title}>
          <Link href={`/products/${lineItem.productSku}`}>
            <a>
              <Image src={lineItem.imageUrl} alt={lineItem.title} className='c-line-items__image' aria-label={lineItem.title} />
              { lineItem.title }
            </a>
          </Link>
        </td>
        <td>
          { lineItem.size }
        </td>
        <td>
          { this.renderOptions(lineItem) }
        </td>
        <td>
          -&pound;{ (lineItem.discount || 0) * lineItem.quantity }
        </td>
        <td>
          &pound;{ lineItem.price * lineItem.quantity }
        </td>
      </tr>
    )
    return cartData
  }

  render () {
    const cart = this.props.cart

    if (cart.lineItems.length === 0) {
      return null
    } else {
      return <table className='c-line-items'>
        <thead>
          <tr className='c-line-items__header'>
            <th scope='col'>
              Your Items
            </th>
            <th scope='col'>
              Size
            </th>
            <th scope='col'>
              Quantity
            </th>
            <th scope='col'>
              You save
            </th>
            <th scope='col'>
              Subtotal
            </th>
          </tr>
        </thead>
        <tbody>
          { this.renderLineItems() }
        </tbody>
      </table>
    }
  }
}
