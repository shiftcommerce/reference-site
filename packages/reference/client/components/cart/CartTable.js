// Libraries
import { Component } from 'react'
import Pluralize from 'react-pluralize'

// Lib
import { fixedPrice } from '../../lib/fixedPrice'
import { calculateCartSummary } from '../../lib/calculateCartSummary'

// Components
import LineItems from './LineItems'
import CartSummary from './CartSummary'
import CartNoData from './CartNoData'

class CartTable extends Component {
  renderCartData () {
    const cart = this.props.cart

    if (cart.totalQuantity === 0) {
      return <CartNoData />
    } else {
      return <section className='u-float-clear'>
        <LineItems {...this.props} aria-label='Line Items' />
        <CartSummary {...this.props} aria-label='Cart Summary' />
      </section>
    }
  }

  render () {
    const { cart } = this.props

    const totals = calculateCartSummary(cart)

    return (
      <section className='c-cart-table c-cart-table--margin'>
        <section className='c-cart-table__header'>
          <div className='u-float--left'>
            <h1 className='c-cart-table__title'> Your Shopping Basket <a className='c-cart-table__amount'>({ cart.totalQuantity })</a> </h1>
            <p className='c-cart-table__description'>You have <a>{ cart.totalQuantity }</a> <Pluralize singular='item' count={cart.totalQuantity} showCount={false} /> in your shopping basket</p>
            <p className='c-cart-table__description'><Pluralize singular='This' plural='These' count={cart.totalQuantity} showCount={false} /> <Pluralize singular='item' count={cart.totalQuantity} showCount={false} /> will be saved for 48 hours depending on availablility</p>
          </div>
          <div>
            <h4 className='c-cart-table__total'>&pound;{ fixedPrice(totals.total) }</h4>
          </div>
        </section>
        { this.renderCartData() }
      </section>
    )
  }
}

export default CartTable
