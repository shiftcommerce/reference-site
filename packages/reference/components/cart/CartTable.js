// Libraries
import { Component } from 'react'

// Components
import LineItems from './LineItems'
import CartSummary from './CartSummary'
import CartNoData from './CartNoData'

class CartTable extends Component {
  renderCartData () {
    const cart = this.props.cart

    if (cart.totalQuantity === 0) {
      return <CartNoData />
    } else {
      return <section className='u-float-clear'>
        <LineItems {...this.props} aria-label='Line Items' />
        <CartSummary {...this.props} aria-label='Cart Summary' />
      </section>
    }
  }
  render () {
    const {
      cart
    } = this.props
    const itemsKey = cart.totalQuantity === 1 ? 'item' : 'items'

    return (
      <section>
        <section className='c-cart-table__header'>
          <div className='u-float--left'>
            <h1> My Bag </h1>
          </div>
          <div className='u-float--right'>
            <h4> <b>{ cart.totalQuantity } { itemsKey }</b> in your bag </h4>
          </div>
        </section>
        { this.renderCartData() }
      </section>
    )
  }
}

export default CartTable
