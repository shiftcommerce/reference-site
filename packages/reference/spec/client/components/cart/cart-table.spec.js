import CartTable from '../../../../client/components/cart/cart-table'
import CartNoData from '../../../../client/components/cart/cart-no-data'
import LineItems from '../../../../client/components/cart/line-items'
import CartSummary from '../../../../client/components/cart/cart-summary'

test('renders the exception messages when no line items are available', () => {
  // arrange
  const cart = {
    lineItems: [],
    totalQuantity: 0
  }

  const updateQuantity = () => {}

  // act
  const wrapper = shallow(
    <CartTable cart={cart} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Your Shopping Basket')
  expect(wrapper).toIncludeText('You have 0')
  expect(wrapper).toContainReact(<CartNoData />)
  expect(wrapper).not.toContainReact(<LineItems cart={cart} updateQuantity={updateQuantity} aria-label='Line Items' />)
  expect(wrapper).not.toContainReact(<CartSummary cart={cart} updateQuantity={updateQuantity} aria-label='Cart Summary' />)
})

test('renders the correct message when line items are available in cart', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        sku: 'S2658934_C101_486',
        variant: 'Regular Fit Wool Mix Trousers',
        quantity: 2,
        stockAvailableLevel: 10000,
        price: '30',
        imageUrl: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/145854/1506433071.3858652-S2658934_C101_Main.jpg',
        productID: '70515',
        canonicalPath: '1',
        slug: '1',
        bundles: [
          {
            id: '261',
            slug: 'anna-outfit-1',
            canonical_path: '/bundles/anna-outfit-1',
            meta_attributes: {},
            name: 'Anna Outfit 1',
            asset_files: [
              {
                id: '157928',
                meta_attributes: {},
                asset_file: '1513767197.7652595-1847-_Anna_Outfit_1-3_099.jpg',
                s3_url: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157928/1513767197.7652595-1847-_Anna_Outfit_1-3_099.jpg',
                canonical_path: 'https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157928/1513767197.7652595-1847-_Anna_Outfit_1-3_099.jpg'
              }
            ]
          }
        ]
      }
    ],
    totalQuantity: 2
  }

  const updateQuantity = () => {}

  // act
  const wrapper = mount(
    <CartTable cart={cart} updateQuantity={updateQuantity} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Your Shopping Basket')
  expect(wrapper).toIncludeText('2 items in your shopping basket')
  expect(wrapper).not.toContainReact(<CartNoData />)
  expect(wrapper).toContainReact(<LineItems cart={cart} updateQuantity={updateQuantity} aria-label='Line Items' />)
  expect(wrapper).toContainReact(<CartSummary cart={cart} updateQuantity={updateQuantity} aria-label='Cart Summary' />)
})
