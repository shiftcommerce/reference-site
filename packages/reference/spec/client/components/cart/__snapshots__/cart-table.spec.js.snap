// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct message when line items are available in cart 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartTable
    cart={
      Object {
        "discount_summaries": Array [],
        "line_items": Array [
          Object {
            "item": Object {
              "product": Object {
                "canonicalPath": "1",
                "slug": "1",
              },
            },
            "sku": "S2658934_C101_486",
            "stock_available_level": 10000,
            "sub_total": 30,
            "title": "Regular Fit Wool Mix Trousers",
            "total": 30,
            "unit_quantity": 2,
          },
        ],
        "line_items_count": 2,
        "sub_total": 60,
      }
    }
    updateQuantity={[Function]}
  >
    <section
      className="c-cart-table"
    >
      <section
        className="c-cart-table__header"
      >
        <div
          className="c-cart-table__breadcrumb"
        >
          <Breadcrumb>
            <nav>
              <ol
                className="o-breadcrumb"
              >
                <li
                  className="o-breadcrumb__item"
                >
                  <Link
                    as="/"
                    href="/home/index"
                  >
                    <a
                      className="o-breadcrumb__link"
                      href="/"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </Link>
                </li>
              </ol>
            </nav>
          </Breadcrumb>
        </div>
        <div
          className="c-cart-table__header-grid"
        >
          <div
            className="c-cart-table__header-grid-item-a"
          >
            <h1
              className="c-cart-table__title"
            >
              Your Shopping Basket 
              <a
                className="c-cart-table__amount"
              >
                (
                2
                )
              </a>
            </h1>
            <div
              className="c-cart-table__text"
            >
              <p
                className="c-cart-table__description"
              >
                You have 
                <a>
                  2
                </a>
                 
                <Plural
                  className={null}
                  count={2}
                  showCount={false}
                  singular="item"
                  style={Object {}}
                  zero={null}
                >
                  <span
                    className={null}
                    style={Object {}}
                  >
                    items
                  </span>
                </Plural>
                 in your shopping basket
              </p>
              <p
                className="c-cart-table__description"
              >
                <Plural
                  className={null}
                  count={2}
                  plural="These"
                  showCount={false}
                  singular="This"
                  style={Object {}}
                  zero={null}
                >
                  <span
                    className={null}
                    style={Object {}}
                  >
                    These
                  </span>
                </Plural>
                 
                <Plural
                  className={null}
                  count={2}
                  showCount={false}
                  singular="item"
                  style={Object {}}
                  zero={null}
                >
                  <span
                    className={null}
                    style={Object {}}
                  >
                    items
                  </span>
                </Plural>
                 will be saved for 48 hours depending on availablility
              </p>
            </div>
          </div>
          <div
            className="c-cart-table__header-grid-item-b"
          >
            <h1
              className="c-cart-table__title"
            >
              Estimated Delivery
            </h1>
            <p>
              We will deliver your item: 
              <a>
                Sunday 2nd December
              </a>
            </p>
          </div>
        </div>
        <div
          className="c-cart-table__header-total"
        >
          <h4
            className="c-cart-table__total"
          >
            Â£
            0.00
          </h4>
        </div>
      </section>
      <section
        className="c-cart-table__grid"
      >
        <div
          className="c-cart-table__grid-item-a"
        >
          <LineItems
            aria-label="Line Items"
            cart={
              Object {
                "discount_summaries": Array [],
                "line_items": Array [
                  Object {
                    "item": Object {
                      "product": Object {
                        "canonicalPath": "1",
                        "slug": "1",
                      },
                    },
                    "sku": "S2658934_C101_486",
                    "stock_available_level": 10000,
                    "sub_total": 30,
                    "title": "Regular Fit Wool Mix Trousers",
                    "total": 30,
                    "unit_quantity": 2,
                  },
                ],
                "line_items_count": 2,
                "sub_total": 60,
              }
            }
            updateQuantity={[Function]}
          >
            <div
              className="c-line-items"
            >
              <div
                className="c-line-items__sections"
                key="1"
              >
                <div
                  className="c-line-items__images"
                >
                  <Link
                    href="/slug?slug=undefined"
                    key="1"
                  >
                    <t
                      className="c-line-items__image"
                      key="1"
                      onClick={[Function]}
                    />
                  </Link>
                </div>
                <div
                  className="c-line-items__information"
                >
                  <div
                    className="c-line-items__title"
                  >
                    <div
                      className="c-line-items__details"
                    >
                      <h4
                        className="c-line-items__details-title  u-bold"
                      >
                        undefined - undefined
                      </h4>
                      <div
                        className="c-line-items__details-sku"
                      >
                        <span>
                          S2658934_C101_486
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    className="c-line-items__colour"
                  >
                    <div
                      className="c-line-items__param  u-bold"
                    >
                      <span>
                        Colour
                      </span>
                    </div>
                  </div>
                  <div
                    className="c-line-items__quantity"
                  >
                    <div
                      className="c-line-items__param u-bold"
                    >
                      <span>
                        Quantity
                      </span>
                    </div>
                  </div>
                  <div
                    className="c-line-items__param  c-line-items__colour-selected"
                  >
                    <span>
                      Grey
                    </span>
                  </div>
                  <div
                    className="c-line-items__param  c-line-items__quantity-selected"
                  >
                    <span>
                      <select
                        onChange={[Function]}
                        value={2}
                      >
                        <option
                          aria-posinset={2}
                          aria-setsize={1}
                          key="1"
                          value={1}
                        >
                          1
                        </option>
                        <option
                          aria-posinset={3}
                          aria-setsize={2}
                          key="2"
                          value={2}
                        >
                          2
                        </option>
                        <option
                          aria-posinset={4}
                          aria-setsize={3}
                          key="3"
                          value={3}
                        >
                          3
                        </option>
                        <option
                          aria-posinset={5}
                          aria-setsize={4}
                          key="4"
                          value={4}
                        >
                          4
                        </option>
                        <option
                          aria-posinset={6}
                          aria-setsize={5}
                          key="5"
                          value={5}
                        >
                          5
                        </option>
                        <option
                          aria-posinset={7}
                          aria-setsize={6}
                          key="6"
                          value={6}
                        >
                          6
                        </option>
                        <option
                          aria-posinset={8}
                          aria-setsize={7}
                          key="7"
                          value={7}
                        >
                          7
                        </option>
                        <option
                          aria-posinset={9}
                          aria-setsize={8}
                          key="8"
                          value={8}
                        >
                          8
                        </option>
                        <option
                          aria-posinset={10}
                          aria-setsize={9}
                          key="9"
                          value={9}
                        >
                          9
                        </option>
                        <option
                          aria-posinset={11}
                          aria-setsize={10}
                          key="10"
                          value={10}
                        >
                          10
                        </option>
                      </select>
                    </span>
                  </div>
                  <div
                    className="c-line-items__remove"
                  >
                    <div
                      className="c-line-items__delete"
                    >
                      <a
                        className="c-line-items__delete-button"
                      >
                        Delete
                      </a>
                    </div>
                  </div>
                  <div
                    className="c-line-items__amounts"
                  >
                    <a
                      className="c-line-items__amount c-line-items__amount--total"
                    >
                      Â£
                      30.00
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </LineItems>
        </div>
        <div
          className="c-cart-table__grid-item-b"
        >
          <Connect(CouponForm)>
            <CouponForm
              dispatch={[Function]}
            >
              <div
                className="c-coupon-form"
              >
                <div
                  aria-label="Use gift card or rewards code"
                  className="c-gift__wrapper"
                >
                  <p
                    className="c-coupon-form__title"
                  >
                    Promotion Code
                  </p>
                  <Formik
                    enableReinitialize={false}
                    initialValues={
                      Object {
                        "couponCode": "",
                      }
                    }
                    isInitialValid={false}
                    onSubmit={[Function]}
                    validate={[Function]}
                    validateOnBlur={true}
                    validateOnChange={true}
                  >
                    <Form>
                      <Component
                        formik={
                          Object {
                            "dirty": false,
                            "errors": Object {},
                            "handleBlur": [Function],
                            "handleChange": [Function],
                            "handleReset": [Function],
                            "handleSubmit": [Function],
                            "initialValues": Object {
                              "couponCode": "",
                            },
                            "isSubmitting": false,
                            "isValid": false,
                            "isValidating": false,
                            "registerField": [Function],
                            "resetForm": [Function],
                            "setError": [Function],
                            "setErrors": [Function],
                            "setFieldError": [Function],
                            "setFieldTouched": [Function],
                            "setFieldValue": [Function],
                            "setFormikState": [Function],
                            "setStatus": [Function],
                            "setSubmitting": [Function],
                            "setTouched": [Function],
                            "setValues": [Function],
                            "submitCount": 0,
                            "submitForm": [Function],
                            "touched": Object {},
                            "unregisterField": [Function],
                            "validate": [Function],
                            "validateField": [Function],
                            "validateForm": [Function],
                            "validateOnBlur": true,
                            "validateOnChange": true,
                            "validationSchema": undefined,
                            "values": Object {
                              "couponCode": "",
                            },
                          }
                        }
                      >
                        <form
                          onReset={[Function]}
                          onSubmit={[Function]}
                        >
                          <div
                            className="c-coupon-form__field"
                          >
                            <FormikConnect(FieldInner)
                              className="c-coupon-form__input"
                              name="couponCode"
                              placeholder="Enter Promotion Code"
                              type="text"
                            >
                              <FieldInner
                                className="c-coupon-form__input"
                                formik={
                                  Object {
                                    "dirty": false,
                                    "errors": Object {},
                                    "handleBlur": [Function],
                                    "handleChange": [Function],
                                    "handleReset": [Function],
                                    "handleSubmit": [Function],
                                    "initialValues": Object {
                                      "couponCode": "",
                                    },
                                    "isSubmitting": false,
                                    "isValid": false,
                                    "isValidating": false,
                                    "registerField": [Function],
                                    "resetForm": [Function],
                                    "setError": [Function],
                                    "setErrors": [Function],
                                    "setFieldError": [Function],
                                    "setFieldTouched": [Function],
                                    "setFieldValue": [Function],
                                    "setFormikState": [Function],
                                    "setStatus": [Function],
                                    "setSubmitting": [Function],
                                    "setTouched": [Function],
                                    "setValues": [Function],
                                    "submitCount": 0,
                                    "submitForm": [Function],
                                    "touched": Object {},
                                    "unregisterField": [Function],
                                    "validate": [Function],
                                    "validateField": [Function],
                                    "validateForm": [Function],
                                    "validateOnBlur": true,
                                    "validateOnChange": true,
                                    "validationSchema": undefined,
                                    "values": Object {
                                      "couponCode": "",
                                    },
                                  }
                                }
                                name="couponCode"
                                placeholder="Enter Promotion Code"
                                type="text"
                              >
                                <input
                                  className="c-coupon-form__input"
                                  name="couponCode"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  placeholder="Enter Promotion Code"
                                  type="text"
                                  value=""
                                />
                              </FieldInner>
                            </FormikConnect(FieldInner)>
                            <t
                              aria-label="Apply Gift Code"
                              className="c-coupon-form__button o-button--sml"
                              containerClassName="c-coupon-form__button-container"
                              label="Apply"
                              status="primary"
                            >
                              <button
                                aria-label="Apply Gift Code"
                                className="c-coupon-form__button o-button--sml o-button o-button--primary"
                                containerClassName="c-coupon-form__button-container"
                                disabled={false}
                                role="button"
                                tabIndex="0"
                              >
                                Apply
                              </button>
                            </t>
                          </div>
                          <FormikConnect(ErrorMessageImpl)
                            className="c-coupon-form__error"
                            component="div"
                            name="couponCode"
                          >
                            <ErrorMessageImpl
                              className="c-coupon-form__error"
                              component="div"
                              formik={
                                Object {
                                  "dirty": false,
                                  "errors": Object {},
                                  "handleBlur": [Function],
                                  "handleChange": [Function],
                                  "handleReset": [Function],
                                  "handleSubmit": [Function],
                                  "initialValues": Object {
                                    "couponCode": "",
                                  },
                                  "isSubmitting": false,
                                  "isValid": false,
                                  "isValidating": false,
                                  "registerField": [Function],
                                  "resetForm": [Function],
                                  "setError": [Function],
                                  "setErrors": [Function],
                                  "setFieldError": [Function],
                                  "setFieldTouched": [Function],
                                  "setFieldValue": [Function],
                                  "setFormikState": [Function],
                                  "setStatus": [Function],
                                  "setSubmitting": [Function],
                                  "setTouched": [Function],
                                  "setValues": [Function],
                                  "submitCount": 0,
                                  "submitForm": [Function],
                                  "touched": Object {},
                                  "unregisterField": [Function],
                                  "validate": [Function],
                                  "validateField": [Function],
                                  "validateForm": [Function],
                                  "validateOnBlur": true,
                                  "validateOnChange": true,
                                  "validationSchema": undefined,
                                  "values": Object {
                                    "couponCode": "",
                                  },
                                }
                              }
                              name="couponCode"
                            />
                          </FormikConnect(ErrorMessageImpl)>
                        </form>
                      </Component>
                    </Form>
                  </Formik>
                </div>
              </div>
            </CouponForm>
          </Connect(CouponForm)>
          <CartSummary
            aria-label="Cart Summary"
            cart={
              Object {
                "discount_summaries": Array [],
                "line_items": Array [
                  Object {
                    "item": Object {
                      "product": Object {
                        "canonicalPath": "1",
                        "slug": "1",
                      },
                    },
                    "sku": "S2658934_C101_486",
                    "stock_available_level": 10000,
                    "sub_total": 30,
                    "title": "Regular Fit Wool Mix Trousers",
                    "total": 30,
                    "unit_quantity": 2,
                  },
                ],
                "line_items_count": 2,
                "sub_total": 60,
              }
            }
            updateQuantity={[Function]}
          >
            <section
              className="c-cart-summary"
            >
              <dl>
                <dt>
                  <a>
                    Total Price:
                  </a>
                </dt>
                <dd>
                  <a>
                    Â£
                    60.00
                  </a>
                </dd>
              </dl>
              <dl>
                <dt>
                  <a>
                    Estimating shipping cost...
                  </a>
                </dt>
              </dl>
              <dl>
                <dt>
                  <p
                    className="u-bold"
                  >
                    TOTAL:
                  </p>
                </dt>
                <dd>
                  <p
                    className="u-bold"
                  >
                    Â£
                  </p>
                </dd>
              </dl>
              <dl>
                <a>
                  * Including VAT
                </a>
              </dl>
            </section>
            <section
              className="c-cart-summary__buttons"
            >
              <Link
                as="/"
                href="/slug?slug=/homepage"
              >
                <a
                  className="o-button--sml c-cart-summary__buttons--continue"
                  href="/"
                  onClick={[Function]}
                >
                  continue shopping
                </a>
              </Link>
              <Link
                href="/checkout"
              >
                <a
                  className="o-button--sml c-cart-summary__buttons--proceed"
                  href="/checkout"
                  onClick={[Function]}
                >
                  Checkout
                </a>
              </Link>
            </section>
          </CartSummary>
          <div
            className="c-cart-table__icons"
          >
            <PaymentIcons>
              <div
                className="c-payment-icons"
              >
                <p>
                  We Accept
                </p>
                <div
                  className="c-payment-icons__images"
                >
                  <t
                    aria-label="VISA"
                    className="c-payment-icons__image"
                    src="/static/payments/visa.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="VISA"
                          className="o-image"
                          src="/static/payments/visa.svg"
                          srcSet="/static/payments/visa.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                  <t
                    aria-label="Mastercard"
                    className="c-payment-icons__image"
                    src="/static/payments/mastercard.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="Mastercard"
                          className="o-image"
                          src="/static/payments/mastercard.svg"
                          srcSet="/static/payments/mastercard.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                  <t
                    aria-label="American Express"
                    className="c-payment-icons__image"
                    src="/static/payments/american-express.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="American Express"
                          className="o-image"
                          src="/static/payments/american-express.svg"
                          srcSet="/static/payments/american-express.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                  <t
                    aria-label="Maestro"
                    className="c-payment-icons__image"
                    src="/static/payments/maestro.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="Maestro"
                          className="o-image"
                          src="/static/payments/maestro.svg"
                          srcSet="/static/payments/maestro.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                  <t
                    aria-label="PayPal"
                    className="c-payment-icons__image"
                    src="/static/payments/pay-pal.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="PayPal"
                          className="o-image"
                          src="/static/payments/pay-pal.svg"
                          srcSet="/static/payments/pay-pal.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                  <t
                    aria-label="Apple Pay"
                    className="c-payment-icons__image"
                    src="/static/payments/apple-pay.svg"
                  >
                    <div
                      className="c-payment-icons__image"
                    >
                      <picture>
                        <img
                          aria-label="Apple Pay"
                          className="o-image"
                          src="/static/payments/apple-pay.svg"
                          srcSet="/static/payments/apple-pay.svg 769w"
                        />
                      </picture>
                    </div>
                  </t>
                </div>
              </div>
            </PaymentIcons>
          </div>
        </div>
      </section>
    </section>
  </CartTable>
</Provider>
`;

exports[`renders the exception messages when no line items are available 1`] = `
<section
  className="c-cart-table"
>
  <section
    className="c-cart-table__header"
  >
    <div
      className="c-cart-table__breadcrumb"
    >
      <Breadcrumb />
    </div>
    <div
      className="c-cart-table__header-grid"
    >
      <div
        className="c-cart-table__header-grid-item-a"
      >
        <h1
          className="c-cart-table__title"
        >
          Your Shopping Basket 
          <a
            className="c-cart-table__amount"
          >
            (
            0
            )
          </a>
        </h1>
        <div
          className="c-cart-table__text"
        >
          <p
            className="c-cart-table__description"
          >
            You have 
            <a>
              0
            </a>
             
            <Plural
              className={null}
              count={0}
              showCount={false}
              singular="item"
              style={Object {}}
              zero={null}
            />
             in your shopping basket
          </p>
          <p
            className="c-cart-table__description"
          >
            <Plural
              className={null}
              count={0}
              plural="These"
              showCount={false}
              singular="This"
              style={Object {}}
              zero={null}
            />
             
            <Plural
              className={null}
              count={0}
              showCount={false}
              singular="item"
              style={Object {}}
              zero={null}
            />
             will be saved for 48 hours depending on availablility
          </p>
        </div>
      </div>
      <div
        className="c-cart-table__header-grid-item-b"
      >
        <h1
          className="c-cart-table__title"
        >
          Estimated Delivery
        </h1>
        <p>
          We will deliver your item: 
          <a>
            Sunday 2nd December
          </a>
        </p>
      </div>
    </div>
    <div
      className="c-cart-table__header-total"
    >
      <h4
        className="c-cart-table__total"
      >
        Â£
        0.00
      </h4>
    </div>
  </section>
  <CartNoData />
</section>
`;
